docs/blog/posts/2024-03-28-colab_gpu.md

```diff
@@ -0,0 +1,236 @@
+Epoch 5/15
+422/422 ━━━━━━━━━━━━━━━━━━━━ 82s 106ms/step - accuracy: 0.9802 - loss: 0.0627 - val_accuracy: 0.9897 - val_loss: 0.0388
+Epoch 6/15
+422/422 ━━━━━━━━━━━━━━━━━━━━ 80s 102ms/step - accuracy: 0.9828 - loss: 0.0590 - val_accuracy: 0.9910 - val_loss: 0.0344
+Epoch 7/15
+422/422 ━━━━━━━━━━━━━━━━━━━━ 44s 105ms/step - accuracy: 0.9838 - loss: 0.0512 - val_accuracy: 0.9907 - val_loss: 0.0322
+Epoch 8/15
+422/422 ━━━━━━━━━━━━━━━━━━━━ 83s 107ms/step - accuracy: 0.9844 - loss: 0.0507 - val_accuracy: 0.9915 - val_loss: 0.0301
+Epoch 9/15
+422/422 ━━━━━━━━━━━━━━━━━━━━ 47s 112ms/step - accuracy: 0.9871 - loss: 0.0419 - val_accuracy: 0.9917 - val_loss: 0.0307
+Epoch 10/15
+422/422 ━━━━━━━━━━━━━━━━━━━━ 45s 106ms/step - accuracy: 0.9866 - loss: 0.0428 - val_accuracy: 0.9918 - val_loss: 0.0293
+Epoch 11/15
+422/422 ━━━━━━━━━━━━━━━━━━━━ 44s 105ms/step - accuracy: 0.9863 - loss: 0.0432 - val_accuracy: 0.9923 - val_loss: 0.0298
+Epoch 12/15
+422/422 ━━━━━━━━━━━━━━━━━━━━ 45s 107ms/step - accuracy: 0.9883 - loss: 0.0361 - val_accuracy: 0.9920 - val_loss: 0.0301
+Epoch 13/15
+422/422 ━━━━━━━━━━━━━━━━━━━━ 82s 107ms/step - accuracy: 0.9884 - loss: 0.0345 - val_accuracy: 0.9928 - val_loss: 0.0282
+Epoch 14/15
+422/422 ━━━━━━━━━━━━━━━━━━━━ 82s 106ms/step - accuracy: 0.9896 - loss: 0.0321 - val_accuracy: 0.9920 - val_loss: 0.0285
+Epoch 15/15
+422/422 ━━━━━━━━━━━━━━━━━━━━ 82s 107ms/step - accuracy: 0.9882 - loss: 0.0360 - val_accuracy: 0.9923 - val_loss: 0.0275
+Test loss: 0.023274529725313187
+Test accuracy: 0.9919000267982483
+```
+
+Enusre that the **paths** for loading data and saving outputs and are correctly set as needed. Below is an example of how to save results, the trained model, and predictions to Google Drive:
+
+```python title="colab_mnist.ipynb"
+import os
+
+# Define the output directory (make sure it exists)
+output_dir = './outputs'
+os.makedirs(output_dir, exist_ok=True)
+
+# Save test results to a file in the output directory
+results_file = os.path.join(output_dir, 'results.txt')
+with open(results_file, 'w') as f:
+   f.write(f"Test loss: {score[0]}\n")
+   f.write(f"Test accuracy: {score[1]}\n")
+print(f"Evaluation results saved to {results_file}")
+
+# Save predictions as a NumPy array in the output directory
+predictions_file = os.path.join(output_dir, 'predictions.npy')
+predictions = model.predict(x_test)
+np.save(predictions_file, predictions)
+print(f"Predictions saved to {predictions_file}")
+
+# Save the entire model in the native Keras format
+model_file = os.path.join(output_dir, 'mnist_model.keras')
+model.save(model_file)
+print(f"Model saved to {model_file}")
+
+```
+
+Running the above code will generate files in your Google Drive, with output messages like this:
+
+```{ .yaml .no-copy title="colab_mnist.ipynb" }
+Evaluation results saved to ./outputs/results.txt
+Predictions saved to ./outputs/predictions.npy
+Model saved to ./outputs/mnist_model.keras
+313/313 ━━━━━━━━━━━━━━━━━━━━ 4s 14ms/step
+```
+
+!!! Tip
```

Comment:
This does not render correctly

Reply:
It looks fine to me. Let me know if it still doesn't render correctly for you.